### DISCUSSION TITLE
Ability to use nicegui elements with a table without creating a custom vue component

### DISCUSSION THREAD

**Discussion by apaulo2004** [zauberzeug/nicegui#2164] [Q&A] [answered]
### Question

I saw someone ask about how one could use a NiceGUI UI element in side of a slot. I am attempting do the same thing based on the table and slots example [](https://github.com/zauberzeug/nicegui/blob/main/examples/table_and_slots/main.py) but I'm wanting to create a body slot and then build it up using the vue structure that should be there using table.row() and table.cell(). Here is the code example I've been trying to get to work.
```py
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = [
    {'id': 0, 'name': 'Alice', 'age': 18},
    {'id': 1, 'name': 'Bob', 'age': 21},
    {'id': 2, 'name': 'Lionel', 'age': 19},
    {'id': 3, 'name': 'Michael', 'age': 32},
    {'id': 4, 'name': 'Julie', 'age': 12},
    {'id': 5, 'name': 'Livia', 'age': 25},
    {'id': 6, 'name': 'Carol'},
]

with ui.table(title='My Team', columns=columns, rows=rows, selection='multiple', pagination=10).classes('w-96') as table:
    with table.add_slot("body"):
        for person in rows:
            with table.row():
                with table.cell():
                    if "id" in person:
                        ui.label(person['id'])
                with table.cell():
                    if "name" in person:
                        ui.label(person['name'])
                with table.cell():
                    if "age" in person:
                        ui.label(person['age'])

ui.run()
```

For some reason this results in each row in the rows to repeat 7 times. The table thinks there is only 7 items.
![table_body](https://github.com/zauberzeug/nicegui/assets/31635177/78a8a1ee-d736-4b65-a244-4ccc0bbdd9fe)
![table_bottom](https://github.com/zauberzeug/nicegui/assets/31635177/26c6279d-a4ea-4dfd-854b-b7b014f593d8)

Any thoughts or suggestions? Ideally I could use something like ui.select() directly instead of having to create a vue template in a slot.

**Comment by falkoschindler** [comment, marked as answer]
Hi @apaulo2004,

These "body" slots are meant to represent any row in the table and usually depend on the current row data. Therefore they are often defined as templates depending on a "props" input variable (as shown in some [demos](https://nicegui.io/documentation/table)). Unfortunately this makes it impossible to define them in terms of NiceGUI elements. At least until now we haven't found a way to write slot templates with NiceGUI elements instead of long multi-line strings. We're still missing an idea how this could work at all. See https://github.com/zauberzeug/nicegui/discussions/2080#discussioncomment-7661491 for a related discussion.

In your specific example you're basically saying that each row should be rendered as 7 rows. Therefore you get 49 rows in total. Furthermore, features like pagination don't work as expected.

**Reply by apaulo2004** [reply]
that makes sense, thanks for clarifying.
